### PR TITLE
[#ENTE-31] Add an export function for visible-service webview

### PR DESCRIPTION
**Pull request by BurnedMarshal**

### CODE REVIEW

File: src/commands/services/export.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+          [_.organizationFiscalCode]: {
+            o: _.organizationName,
+            fc: _.organizationFiscalCode,
+            s: [serviceMapper(_)]
+          }
+        };
+      }, {} as Record<string, ServicesExport>);
+      cli.action.stop();
+      // tslint:disable-next-line: no-console
+      console.log(
```
Comment (pasqualedevita) [inline src/commands/services/export.ts:104]: PR it's ok for me. 
Do u try to export all services? In some string field, such as description, I found double quotes or other chars that can break the json.

**Review by pasqualedevita — APPROVED** [review]

File: src/commands/services/export.ts (hunk 1 above)
Comment (BurnedMarshal) [reply src/commands/services/export.ts:104]: Yes, the `JSON.stringify` conversion backquotes the double-quotes.

File: src/commands/services/export.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        JSON.stringify(
+          Object.values(organizationMap).sort((a, b) => {
```
Comment (Undermaken) [inline src/commands/services/export.ts:106]:
```suggestion
          Object.values(organizationMap).sort((a, b) => a.o.toLowerCase().localeCompare(b.o.toLowerCase())
```
Could it be a simplification?

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: src/commands/services/export.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+// tslint:disable-next-line: no-submodule-imports
+import { RetrievedService } from "io-functions-commons/dist/src/models/service";
+import { DateTime } from "luxon";
+import { ServiceExport, ServicesExport } from "../../definitions/ServiceExport";
+import { ServiceScopeEnum } from "../../generated/ServiceScope";
+import { getServices } from "../../utils/service";
+
+dotenv.config();
+
+enum ScopeFilter {
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply src/commands/services/export.ts]: Nice catch. Fixed in d435c83
A little refractory will still be necessary in [here](https://github.com/pagopa/io-ops/pull/132/commits/d435c836aef4b0419ef5d38b707ed959d64d2625#diff-1dc4d2991e24bc4d0327cf99c2fdf1f023a3615cd49d5cb42129353664f5bb5eR73)

File: src/commands/services/export.ts (hunk 2 above)
Comment (BurnedMarshal) [reply src/commands/services/export.ts]: Fixed in c36e77c

File: README.md [hunk 4]
```diff
@@ -386,6 +387,24 @@ OPTIONS
...
+
+```
+USAGE
+  $ io-ops services:export
+
+OPTIONS
+  -s, --scope=ALL|LOCAL|NATIONAL  [default: ALL] The service scope metadata
+  -x, --extended                  show extra columns
+
+  --day=day                       filter services from specified day (Europe/Rome timezone, required format yyyy-MM-dd,
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply README.md]: Fixed in f2af8ec

File: src/commands/services/export.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+          );
+        }
+        return true;
+      });
+      cli.action.stop();
+
+      cli.action.start("Remap services...");
+      const serviceMapper = getServiceMapper(parsedFlags.extended);
+      const organizationMap = visible.reduce((prev, _) => {
+        if (prev[_.organizationFiscalCode]) {
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply src/commands/services/export.ts:80]: To do this change we need a refactory of the output data structure or we lose the information about different fc. Could be this a problem?
```javascript
o: "Organization Name",
fc: "fiscal_code",
s: services
```

[the remark this answers is not in the document]
Comment (Undermaken) [reply src/commands/services/export.ts:80]: I don't know. Maybe we could keep this data structure and change the logic on how this data will be shown on UI side. What do you think?

[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply src/commands/services/export.ts:80]: Our goal should be to minify the computation time client-side. Maybe is fiscal code used to show the proper organization logo? cc. @vito80ba

[the remark this answers is not in the document]
Comment (Undermaken) [reply src/commands/services/export.ts:80]: Yes it is
https://github.com/pagopa/io-services-metadata/tree/master/logos/organizations#how-to-create-a-new-logo
I'm looking for the current web implementation and it seems @vito80ba already does this grouping in the right way.
@vito80ba isn't is so?
### web UI
Schermata 2021-03-26 alle 11 47 43
### services
Schermata 2021-03-26 alle 11 48 41

[the remark this answers is not in the document]
Comment (vito80ba) [reply src/commands/services/export.ts:80]: yes, the fiscal-code is used only for display the proper logo.

**Review by Undermaken — APPROVED** [review]

**Pull request merged**